plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/PartitionedFileSystemLayout.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spooling.filesystem;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.hash.HashCode;
+import com.google.common.hash.Hashing;
+import io.trino.filesystem.Location;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class PartitionedFileSystemLayout
+        extends SimpleFileSystemLayout
+{
+    private final int partitions;
+
+    public PartitionedFileSystemLayout(PartitionedLayoutConfig layoutConfig)
```

Comment:
do we really need that config?
I would just got with what is reasonable for S3. Probably you also want to denote that this layout is for S3 purposes in this name. It is not only partitioning which is important here, but how paths are built.